imap-proto/src/types/acls.rs

```diff
@@ -0,0 +1,259 @@
+    /// mailbox)
+    Lookup,
+    /// r - read (SELECT the mailbox, perform STATUS)
+    Read,
+    /// s - keep seen/unseen information across sessions (set or clear
+    /// \SEEN flag via STORE, also set \SEEN during APPEND/COPY/
+    /// FETCH BODY[...])
+    Seen,
+    /// w - write (set or clear flags other than \SEEN and \DELETED via
+    /// STORE, also set them during APPEND/COPY)
+    Write,
+    /// i - insert (perform APPEND, COPY into mailbox)
+    Insert,
+    /// p - post (send mail to submission address for mailbox,
+    /// not enforced by IMAP4 itself)
+    Post,
+    /// k - create mailboxes (CREATE new sub-mailboxes in any
+    /// implementation-defined hierarchy, parent mailbox for the new
+    /// mailbox name in RENAME)
+    CreateMailbox,
+    /// x - delete mailbox (DELETE mailbox, old mailbox name in RENAME)
+    DeleteMailbox,
+    /// t - delete messages (set or clear \DELETED flag via STORE, set
+    /// \DELETED flag during APPEND/COPY)
+    DeleteMessage,
+    /// e - perform EXPUNGE and expunge as a part of CLOSE
+    Expunge,
+    /// a - administer (perform SETACL/DELETEACL/GETACL/LISTRIGHTS)
+    Administer,
+    /// n - ability to write .shared annotations values
+    /// From RFC 5257
+    Annotation,
+    /// c - old (deprecated) create. Do not use. Read RFC 4314 for more information.
+    OldCreate,
+    /// d - old (deprecated) delete. Do not use. Read RFC 4314 for more information.
+    OldDelete,
+    /// A custom right
+    Custom(char),
+}
+
+impl From<char> for ACLRight {
+    fn from(c: char) -> Self {
+        match c {
+            'l' => ACLRight::Lookup,
+            'r' => ACLRight::Read,
+            's' => ACLRight::Seen,
+            'w' => ACLRight::Write,
+            'i' => ACLRight::Insert,
+            'p' => ACLRight::Post,
+            'k' => ACLRight::CreateMailbox,
+            'x' => ACLRight::DeleteMailbox,
+            't' => ACLRight::DeleteMessage,
+            'e' => ACLRight::Expunge,
+            'a' => ACLRight::Administer,
+            'n' => ACLRight::Annotation,
+            'c' => ACLRight::OldCreate,
+            'd' => ACLRight::OldDelete,
+            _ => ACLRight::Custom(c),
+        }
+    }
+}
+
+impl From<ACLRight> for char {
+    fn from(right: ACLRight) -> Self {
+        match right {
+            ACLRight::Lookup => 'l',
+            ACLRight::Read => 'r',
+            ACLRight::Seen => 's',
+            ACLRight::Write => 'w',
+            ACLRight::Insert => 'i',
+            ACLRight::Post => 'p',
+            ACLRight::CreateMailbox => 'k',
+            ACLRight::DeleteMailbox => 'x',
+            ACLRight::DeleteMessage => 't',
+            ACLRight::Expunge => 'e',
+            ACLRight::Administer => 'a',
+            ACLRight::Annotation => 'n',
+            ACLRight::OldCreate => 'c',
+            ACLRight::OldDelete => 'd',
+            ACLRight::Custom(c) => c,
+        }
+    }
+}
+
+#[derive(Debug, Eq, PartialEq)]
+pub struct ACLRightList {
```

Comment:
It's not obvious that we should have this type inside imap-proto. I think we should instead use a `Vec<AclRight>` which directly reflects what the server sent us.

Reply:
Ok.. I factored this out